lnwallet/channel.go

```diff
@@ -272,7 +273,7 @@ func (c *commitment) toChannelDelta(ourCommit bool) (*channeldb.ChannelDelta, er
 	}
 
 	for _, htlc := range c.outgoingHTLCs {
-		if htlc.isDust {
+		if (ourCommit && htlc.isDustLocal) || (!ourCommit && htlc.isDustRemote) {
```

Comment:
If it is dust, we don't want to go looking for a non-existant output in the commit tx. If the error in `locateOutputIndex` is removed we can avoid this and just let the HTLC run through it without finding anything and return 0, but I thought it was nicer to have dust-outputs be MaxUint16.